src/Microsoft.Diagnostics.Monitoring.WebApi/Packaging.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Packaging;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+
+namespace Microsoft.Diagnostics.Monitoring.WebApi
+{
+    /// <summary>
+    /// WARNING! This is a temporary Zip based implementation for diagsession files, but likely not the way
+    /// these will be created in the long term.
+    /// </summary>
+    internal static class Packaging
+    {
+        private static readonly XNamespace Ns = "urn:diagnosticshub-package-metadata-2-1";
+
+        public static async Task<string> CreateDiagSession(string dumpFilePath, string dac, string dbi, CancellationToken token)
+        {
+            string packagePath = Path.ChangeExtension(dumpFilePath, ".diagsession");
+
+            //Note it is critical to have Create/Write here. A Read/Write mode will cause
+            //the entire dump to be placed into MemoryStream
+            using Package package = Package.Open(packagePath, FileMode.Create, FileAccess.Write);
+
+            Guid tool = new Guid("013914d7-d262-4da2-8fa4-ba8b4e328a1e");
```

Comment:
Just to confirm because I haven't seen this before: did we generate a GUID so we would have a unique identifier, but then we manually specify that GUID for consistency instead of always producing new GUIDs?

Reply:
When VS will consume this file, it will attempt to use the specific tool. The guid of the tool is fixed.